.github/workflows/multios-unit-tests.yml

```diff
@@ -51,10 +45,16 @@ jobs:
         shell: pwsh
         run: |
           "normalized_workspace=${{ github.workspace }}" >> $env:GITHUB_ENV
+      - name: Restore repo cache
+        uses: actions/cache@0400d5f644dc74513175e3cd8d07132dd4860809 # v4.2.4
+        with:
+          path: .git
+          key: gitdb-${{ github.repository_id }}-${{ github.sha }}
```

Comment:
Why is this necessary? How is downloading the cache any faster than downloading the repo itself?

Maybe consider doing a shallow-clone instead?